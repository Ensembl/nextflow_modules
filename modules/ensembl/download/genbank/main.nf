// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

process DOWNLOAD_GENBANK {
    tag "${meta.production_name}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "ensemblorg/ensembl-genomio:GenomioDockerRebuild_v1.5.0a"

    input:
        val(meta)

    output:
        tuple val(meta), path("output.gb"), emit: gb_sequence
        path "versions.yml" , emit: versions // need to add versions output to other ported modules!

    when:
        task.ext.when == null || task.ext.when

    script:
        def prefix = task.ext.prefix ?: "${meta.accession}"
        output_file = "output.gb"

        """
        genbank_download --accession ${meta.accession} --output_file ${output_file} --debug

        echo -e -n "${task.process}:\n\tensembl-genomio: " > versions.yml
        python -c "import ensembl.io.genomio; print(ensembl.io.genomio.__version__)" >> versions.yml
        """

    stub:
        def prefix = task.ext.prefix ?: "${meta.accession}"
        output_file = "output.gb"
        """
        cp ${workflow.projectDir}/tests/modules/ensembl/download/genbank/${output_file} ./${output_file}
        
        echo -e -n "${task.process}:\n\tensembl-genomio: " > versions.yml
        python -c "import ensembl.io.genomio; print(ensembl.io.genomio.__version__)" >> versions.yml
        """
}
