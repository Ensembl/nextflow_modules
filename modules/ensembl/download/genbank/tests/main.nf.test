// See the NOTICE file distributed with this work for additional information
// regarding copyright ownership.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// nf-core modules test download/genbank
nextflow_process {

    name "Test Process DOWNLOAD_GENBANK"
    script "../main.nf"
    process "DOWNLOAD_GENBANK"

    tag "modules"
    tag "modules_nfcore"
    tag "download"
    tag "download/genbank"

    test("Download genbank [stub]") {
        when {

            options "-stub"

            process {
                """
                input[0] = [[ accession:'test']]
                """
            }
        }

        then {
            assert snapshot(
                    process.out.gb_sequence.collect {
                        it.collect { it instanceof String && file(it).exists() ? file(it).name : it }
                        }
                    ).match("genabank_seq_record")
            
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.gb_sequence.size() == 1 },
                { assert process.out.gb_sequence != null }
            
            )
        }
    }
}
