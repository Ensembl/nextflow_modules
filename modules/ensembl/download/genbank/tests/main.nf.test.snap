{
    "Download genbank [stub]": {
        "content": [
            {
                "0": [
                    [
                        [
                            {
                                "accession": "test"
                            }
                        ],
                        "output.gb:md5,7aad1ec58b439ead323ea7b5ab7eaddb"
                    ]
                ],
                "1": [
                    "versions.yml:md5,47ac8e6a4decc2bdfb4550b2b44e96c7"
                ],
                "gb_sequence": [
                    [
                        [
                            {
                                "accession": "test"
                            }
                        ],
                        "output.gb:md5,7aad1ec58b439ead323ea7b5ab7eaddb"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,47ac8e6a4decc2bdfb4550b2b44e96c7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-12-19T16:23:43.300514591"
    },
    "genabank_seq_record": {
        "content": [
            [
                [
                    [
                        {
                            "accession": "test"
                        }
                    ],
                    "output.gb"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-12-02T15:31:47.170791005"
    }
}